indico/modules/events/registration/models/forms.py

```diff
@@ -406,6 +407,8 @@ def is_modification_allowed(self, registration):
             return False
         elif self.modification_mode == ModificationMode.allowed_always:
             return True
+        elif self.modification_mode == ModificationMode.allowed_until_approved:
+            return not registration.state == RegistrationState.complete
```

Comment:
there's `!=` ;)

Also, I think this setting should explicitly check for the `pending` state.